libs/service-logging/src/service_logging/middleware.clj

```diff
@@ -66,16 +71,36 @@
   [handler]
   (fn [request]
     (log-request request)
+    (let [response (handler request)]
+      (log-response request response)
+      response)))
+
+(defn wrap-response-logging
+  "Logs incoming requests and their responses with the 'AccessLogger' logger.
+
+    If the response map contains a `throwable` key, the value will be given to
+    the logger as an exception/throwable. Also, if the `throwable` key is not nil,
+    it is assumed that there will also be an `exception` key in the response map.
+    The `throwable` and `exception` keys are not logged nor passed with the response."
+  [handler]
+  (fn [request]
     (let [{:keys [throwable exception] :as response} (handler request)]
       (if (nil? throwable)
                (log-response request response)
                (tc/with-logging-context {:exception (cheshire/encode exception)}
-                                        (log-response :error throwable request (dissoc response
-                                                                                       :throwable
-                                                                                       :exception))))
-      (dissoc response
-              :throwable
-              :exception))))
+                                        (log-response :error throwable request response)))
+      response)))
+
+(defn add-user-to-context
+  "add-user-to-context is a ring handler that adds the user value from the query
+  string into the context with a key of 'user'. The query params need to be parsed first."
+  [handler]
+  (fn [request]
+    (let [q-params (:query-params request)]
```

Comment:
I think you could use destructuring here to get rid of this `let`:
```clojure
(fn [{:keys [query-params] :as request}]
...
```
Then just use `query-params` below instead of `q-params` or `(:query-params request)`.